src/environmentbase/template.py

```diff
@@ -537,6 +506,76 @@ def add_cloudwatch_alarm(self,
 
         return self.add_resource(alarm)
 
+    @staticmethod
+    def image_map_name(image_label):
+        """
+        Each image map is a concatenation of a custom label and a prefix. This function
+        provides a consistant way to get the correct image map name from the custom label.
+        @returns string
+        """
+        return Template.IMAGE_MAP_PREFIX + image_label.title()
+
+    def get_instancetype_param(self, instance_type, image_name, layer_name):
+        """
+        @param instance_type [string] Default instance_type value.
+        @param image_name [string] Key from config.image_map, used to create Parameter name
+        @param layer_name [string] Functional name for the template, also used in Parameter name
+        Returns a troposphere.Parameter configured to take instance types.  The parameter is only
+        created if it does not already exist.
+        """
+        if not isinstance(instance_type, basestring):
+            raise Exception('Template.get_instancetype_param::instance_type should be string')
+
+        param_name = layer_name.title() + 'InstanceTypeFor' + image_name.title()
```

Comment:
Same comment as above: `I think it would be better to leave .title() off, cause it's hard to reverse engineer the image_label after it has been titled.`